src/shapes/TextPath.ts

```diff
@@ -105,6 +107,21 @@ export class TextPath extends Shape<TextPathConfig> {
     this._setTextData();
   }
 
+  _readDataAttribute() {
+    this.dataArray = Path.parsePathData(this.attrs.data);
+    // in case document is not defined (server side rendering)
+    // use the KonvaJs Path class instead of the browser's native SVGPathElement
+    if (typeof window !== 'undefined' && document) {
+      this.path = document.createElementNS(
```

Comment:
That is interesting. How hard may it be to replicate native API. So we have some calculations in place, but they are not perfect, right? Are there any libs that may help?

Reply:
With your last fixes it still doesn't work good as native?

Reply:
This PR fixed the alignment problem, but this is a different one. 

Approximation accuracy is not crucial for this fix. Using Konva `Path.getPointAtLength` and `Path.getLength` works ok-ish, but when the path has a cubic or a quadratic Bezier curve, the letter spacing between two glyphs is inconsistent, cause the length between two points is not accurate.

I haven't touched the `Path` methods. The problem is in `Path.getPointAtLength`, probably somewhere in `getPointOnCubicBezier` or `getPointOnQuadraticBezier`. Perhaps, browsers are not using De Casteljau's algorithm, but it's just a guess.

I'm not aware of any libs having similar to the native SVG `getPointAtLength` method.

Reply:
I found one polyfill [svg-path-properties](https://github.com/rveciana/svg-path-properties). WDYT about fixing `getPointAtLength` as a follow up PR? I prefer to keep this pull request about TextPath only.